parachain/primitives/xcm-support/src/lib.rs

```diff
@@ -91,39 +90,37 @@ where
 	T::AccountId: AsRef<[u8; 32]>,
 {
 	fn reserve_transfer(
-		origin: <T as frame_system::Config>::Origin,
 		asset_id: SnowbridgeAssetId,
 		para_id: u32,
 		recipient: &T::AccountId,
 		amount: U256,
 	) -> frame_support::dispatch::DispatchResult {
-		let origin_location = T::ExecuteXcmOrigin::ensure_origin(origin.clone())?;
-
 		let amount = u128::try_from(amount).map_err(|e| DispatchError::Other(e))?;
 
+		let origin_location: MultiLocation = MultiLocation {
+			parents: 0,
+			interior: Junctions::X1(Junction::AccountId32 {
+				network: NetworkId::Any,
+				id: recipient.as_ref().clone(),
+			}),
+		};
+
 		let mut remote_message = Xcm(vec![
 			BuyExecution {
 				fees: MultiAsset {
-					id: AssetId::Concrete(MultiLocation { parents: 0, interior: Junctions::Here }),
-					fun: Fungibility::Fungible(0),
-				},
-				weight_limit: Limited(0),
-			},
-			DepositAsset {
-				assets: Wild(All),
-				max_assets: 2,
-				beneficiary: MultiLocation {
-					parents: 0,
-					interior: Junctions::X1(Junction::AccountId32 {
-						network: NetworkId::Any,
-						id: recipient.as_ref().clone(),
-					}),
+					id: Concrete(MultiLocation { parents: 1, interior: Junctions::Here }),
+					fun: Fungible(4_000_000),
 				},
+				weight_limit: Limited(4_000_000),
 			},
+			DepositAsset { assets: Wild(All), max_assets: 2, beneficiary: origin_location.clone() },
 		]);
 
-		let remote_weight: u64 = T::Weigher::weight(&mut remote_message)
-			.map_err(|_| DispatchError::Other("Unweighable message."))?;
+		const WEIGHT_MULTIPLIER: u64 = 2;
```

Comment:
Need to figure out a better way of calculating remote weight. 

We cannot use the local weight of the message because they might be different per parachain. For instance the test parachain uses double the weight. This is probably because we have tuned weights on the snowbridge parachain vs the default weights in the test parachain.

Reply:
Indeed, yeah this problematic. What I think we need to do is have the user/client pass the fee amount (in DOT) for `BuyExecution` on the Ethereum side, as a parameter to `EthApp.lock`. And then we use `WeightLimit::Unlimited` when constructing the `BuyExecution` instruction.

That way we bypass the problem, and make it the client's responsibility to figure out the necessary fees for remote execution.

Hopefully there will be some sort of improved negotiation protocol in XCM so that we can introspect the weight of remote chains.

Reply:
yep agree, should be client's responsibility